boards/arm/samv7/common/src/Make.defs

```diff
@@ -20,9 +20,7 @@
 
 ifeq ($(CONFIG_ARCH_BOARD_COMMON),y)
 
-ifeq ($(CONFIG_BOARDCTL_RESET),y)
 CSRCS += sam_reset.c
-endif
```

Comment:
@michallenc keep it here will speed up the compilation time, because the precompiler will not analyze this file at all if CONFIG_BOARDCTL_RESET is not enabled

Reply:
Yes, but I might want `sam_reset.c` even if `CONFIG_BOARDCTL_RESET` is not defined and `CONFIG_BOARDCTL_RESET_CAUSE` is used only (which is a legit case). We could split `CONFIG_BOARDCTL_RESET_CAUSE` and `CONFIG_BOARDCTL_RESET` into separate files but it tends to make the code less comprehensive.

Reply:
@acassis do you have more concern?